modules/ethereum/src/lib.rs

```diff
@@ -893,17 +893,40 @@ pub fn verify_transaction_finalized<S: Storage>(
 	proof: &[(RawTransaction, RawTransactionReceipt)],
 ) -> bool {
 	if tx_index >= proof.len() as _ {
+		frame_support::debug::trace!(
+			target: "runtime",
+			"Tx finality check failed: transaction index ({}) is larger than number of transactions ({})",
+			tx_index,
+			proof.len(),
+		);
+
 		return false;
 	}
 
 	let header = match storage.header(&block) {
 		Some((header, _)) => header,
-		None => return false,
+		None => {
+			frame_support::debug::trace!(
+				target: "runtime",
+				"Tx finality check failed: can't find header in the storage: {}",
+				block,
+			);
+
+			return false;
+		}
 	};
 	let finalized = storage.finalized_block();
 
 	// if header is not yet finalized => return
 	if header.number > finalized.number {
+		frame_support::debug::trace!(
+			target: "runtime",
+			"Tx finality check failed: header {} is not finalized: {} > {}",
```

Comment:
```suggestion
			"Tx finality check failed: header {} (#{}) - best finalized: {}",
```

Or something like this. I don't think it would be clear from the logs what `{} > {}` means